### ISSUE TITLE
Can't Login into database anymore

### ISSUE THREAD

**Issue by emacattack2020** [MacPass/MacPass#976]
**Describe the bug**
I've been logging into macpass for years now.  But today it will not let me login.  I'm absoultely sure the password is correct.  I went to my backup kdbx files and they won't unlock either.

I get the following error:

The password and/or the keyfile is wrong (1017)

I've attacked a screenshot

The macpass version is 0.7.3
macos is: macos Mojave 10.14.6

<img width="1435" alt="Screen Shot 2019-09-20 at 9 51 25 AM" src="https://user-images.githubusercontent.com/55592817/65336917-d19ad380-db8c-11e9-9d48-3a1d0f80fe25.png">

**Comment by mstarke** [comment]
0.7.3 is quite old, can you try to use the current version (0.7.9) instead? If you are sure you do not have changed the database and the key is correct, could you try to open the file with another client (e.g. KeePassXC)?

**Comment by emacattack2020** [comment]
I downloaded 0.7.9 and am still not able to unlock the database.  I'm sure the password is correct.  How can i tell if i changed the database?  I just have been clicking on the same kdbx file all these years and it opens

<img width="1439" alt="Screen Shot 2019-09-20 at 10 06 14 AM" src="https://user-images.githubusercontent.com/55592817/65337770-723dc300-db8e-11e9-9b79-5241ad2a9d0d.png">

**Comment by mstarke** [comment]
Try another client, if any other client can open the file then this seems to be a problem with MacPass which need investigation. Could you give https://keepassxc.org/download/#mac a try?

As for backups, if those aren't working either this points even more to a bug in MacPass.

**Comment by emacattack2020** [comment]
I just tried KeePassXC and i got the following:
<img width="826" alt="Screen Shot 2019-09-20 at 10 11 01 AM" src="https://user-images.githubusercontent.com/55592817/65338072-0f006080-db8f-11e9-8a59-48ab9b47bca1.png">

**Comment by mstarke** [comment]
This points to either a corrupt database or indeed invalid credentials. Can you try a backup file?

**Comment by DominusExult** [comment]
Or a missing key file? That's been a user error before

**Comment by emacattack2020** [comment]
I've tried two of my backup files to no avail

**Comment by emacattack2020** [comment]
Missing key?

**Comment by mstarke** [comment]
When was the last time you were able to open the file and when was the last backup created?
It seems quite improbable that all of you backups got corrupted as well.

**Comment by emacattack2020** [comment]
i opened it up yesterday.  My backup files were from the Sept 18th

**Comment by mstarke** [comment]
If the files are corrupt or you do not know the password anymore then there is nothing to do about it. The only backup might be to use a windows machine and let KeePass open the file in repair mode, which might help you fix the issue. But - sadly for you - there is no way to recover a lost key

**Comment by emacattack2020** [comment]
In the future, would your recommend exporting the database to an xml file and saving it a safe place for just such an emergency?

**Comment by emacattack2020** [comment]
i do have time machine that has backups from weeks ago.  i will try that

**Comment by emacattack2020** [comment]
what do you mean by lost key?

**Comment by DominusExult** [comment]
Are you sure you didn't use a keyfile? It's a file with some gibberish in it that is used as an added security with the password (so the password alone doesn't unlock). If you deleted that file accidentally you cannot unlock your database.

And make sure the password is written correctly. Just maybe your keyboard layout changed and while you think you enter it correctly it is written differently. Type it in a textfile and copy/paste it.

**Comment by emacattack2020** [comment]
i never created a key file.  would macpass create that somewhere for me?  because ive never seen it.  

i brought out my password on paper and verified it is correct.  however i also have it on a flash drive.  so i will copy and paste that just to be sure

**Comment by emacattack2020** [comment]
as of right now the consensus is either wrong credentials or corrupted database?  

if going back in my time machine a few weeks or even months (when i knew it worked) still has same problem then are we saying it has to be credentials?

**Comment by DominusExult** [comment]
MacPass doesn't create it unless you tell it to. There have been people who created that without completely understanding it and then forgot about it.

**Comment by DominusExult** [comment]
And yes, if your old backups don't work anymore the problem lies not with MacPass

**Comment by emacattack2020** [comment]
if i did create a key file then i would have given it a name and saved it in a location myself then?  i don’t ever remember doing that so im guessing that is not the cause

**Comment by emacattack2020** [comment]
First of all I want to thank everyone who helped me troubleshoot this.  Your support was invaluable and your insight was right on.  I went back thru my time machine and was still unable to open the database.  So this definitely meant it had to be an authentication issue.  Thankfully Dominus suggested to copy and paste my password which i have on a flash drive and I was able to login successfully!!!  For some reason I kept forgetting to add one character to my password even when I was looking right at it on a piece of paper.  Very embarrassing,  but I'd rather be embarrassed than be locked out of macpass.  So to conclude I still love macpass!  It's a great application and I do love the support you all give.   And just feel silly at this point for forgetting my password.  Cheers!
